mewarpx/mewarpx/emission.py

```diff
@@ -0,0 +1,867 @@
+                raise ValueError("Conductor WF must be set for emitters.")
+        else:
+            warnings.warn("No conductor set for emitter. Power will not be "
+                          "correct.")
+
+        self.use_Schottky = use_Schottky
+        self.emission_type = emission_type
+
+        if self.use_Schottky:
+            raise NotImplementedError("Schottky enhancement not implemented"
+                                      " in WarpX yet.")
+
+    def getvoltage(self):
+        if self.conductor is None:
+            return 0.
+
+        return self.conductor.getvoltage()
+
+    def getvoltage_e(self):
+        """Electrical voltage includes WF, eg the Fermi level voltage."""
+        if self.conductor is None:
+            return 0.
+
+        return self.conductor.getvoltage() + self.conductor.WF
+
+    def apply_Schottky_weights(self, particle_dict):
+        """Variable weight function for field-enhanced Schottky emission.
+        Arguments:
+            particle_dict (dict): Particle dictionary returned by
+                get_newparticles() during standard thermionic injection. Any
+                existing variable weights in the dictionary should be
+                normalized to match the total zero-field saturation current of
+                the Emitter without any Schottky enhancement.
+        Returns:
+            new_weights (np.ndarray): 1D array of updated particle weights.
+        Notes:
+            This function requires the "T" attribute of the Emitter to be set
+            for specifying the emitter temperature in Kelvin, and also requires
+            the Emitter subclass to implement a get_normals() function.
+        """
+        raise NotImplementedError
+
+    def get_normals(self, x, y, z):
+        """Calculate local surface normal at specified coordinates.
+        Arguments:
+            x (np.ndarray): x-coordinates of emitted particles (in meters).
+            y (np.ndarray): y-coordinates of emitted particles (in meters).
+            z (np.ndarray): z-coordinates of emitted particles (in meters).
+        Returns:
+            normals (np.ndarray): nx3 array containing the outward surface
+                normal vector at each particle location.
+        """
+        raise NotImplementedError('Normal calculations must be implemented by'
+                                  + ' Emitter sub-classes.')
+
+class ZPlaneEmitter(Emitter):
+
+    """This is the standard injection for a planar cathode."""
+    geoms = ['Z', 'XZ', 'XYZ']
+
+    def __init__(self, conductor, T, xmin=None, xmax=None,
+                 ymin=None, ymax=None, transverse_fac=1.0, **kwargs):
+        """Initialize an emitter for a planar cathode.
+
+        Arguments:
+            conductor (warp.Assembly): Conductor object, used to obtain work
+                function. Can later grab other variables from this conductor.
+            T (float): Temperature in Kelvin for the emitter; determines
+                velocities.
```

Comment:
Can we set this to None by default and grab it from the conductor? We can still have it as a parameter in case we want to inject different particles with different Ts from the same conductor, but that should be a minority use-case...